### DISCUSSION TITLE
A few questions

### DISCUSSION THREAD

**Discussion by ghost** [helloSystem/hello#556] [Q&A] [answered]
I'm not quite sure how to get started, but I'd like to poke around with how helloSystem works.

I've a few questions, so please bear with me:
Firstly, I'm not sure how the system GUI is drawn (for example, default elements like buttons and textareas) - my guess would be somewhere in BreezeEnhanced? I'd like to know for sure.

Secondly, I'm completely lost as to what creating a live ISO does. Does it create an ISO of the current operating system, or does it make a redistributable ISO that has the same setup phase as the ISO on the releases page?

Thirdly, I'm having some problems running helloSystem in VirtualBox (host OS Windows), where the preview OS (before actually installing) takes a very long time to boot (but, in the end, does). After installing helloSystem (in VirtualBox), I'm prompted to restart, which I do. This hangs the VM with a grey-and-black screen for hours. If I kill the VM during this hang and restart it, I end up with the same "install helloSystem" process. I could try running helloSystem on bare metal (Intel i7), but I don't believe I'm enough of a linux user to manage pulling that off, so any help with the VirtualBox install would be helpful.

edit:
another question: is all of the system setup stuff and such managed through pyqt5 `.app`s? I found things like LanguageSelect.app, which I believe is for the system setup process... where does this get started/what does it get started by?

**Comment by NitroNils** [comment]
Good news: You don't have to be a Linux user at all! Jokes aside. I urge you to simply install it on bare metal. Simply follow the instructions and test. As long as you have no important files or backing up the system which you install on there can be no harm in trying. I find it very satisfying. Never enjoyed VMs.

Just my two cents. Hopefully someone else can help you with your questions.

**Comment by louies0623** [comment]
The developers of the system have said that there will be many opportunities to encounter "problems" when the system is executed on a virtual machine, so it has been recommended that users use a physical machine.

**Reply by ghost** [reply]
Ok, I'll try that.

**Reply by louies0623** [reply]
It is recommended to use balenaEtcher to make the downloaded ISO file into a Live USB.

**Reply by probonopd** [reply]
And once you are running helloSystem, it has a built-in "Create Live Media" Utility.

**Reply by ghost** [reply]
Ok, thanks. Is there a way to install helloSystem without erasing my hard drive?

**Reply by probonopd** [reply]
No. But you can try it by running it from Live Media without installation.

**Reply by ghost** [reply]
hrm...
after installing hellosystem, how do I actually contribute to it? what I mean is, how do change/test/publish things? (such as the buttons - where are they drawn from? I've done some more looking and it seems like its a Qt thing, but surely Qt has a way of changing how the buttons look?). also, what does helloSystem use for window management/desktop environment? is it its own thing, kwin(.app), or something else?

**Reply by louies0623** [reply]
Only developers know this, but I can provide the design file of the theme for your reference.
https://github.com/helloSystem/Artwork/issues/5#issue-1480056527

Maybe the system's theme processing is here https://github.com/helloSystem/QtPlugin/issues/1

**Reply by ghost** [reply]
that's exactly what I was looking for! thank you!

sorry for asking so many questions, I just don't really understand how everything comes together (but I think I get how everything works on its own)

**Reply by probonopd** [reply]
Please check out https://hellosystem.github.io/docs/developer/contributing.html

The various projects are hosted at https://github.com/helloSystem/. Glad you are here!

**Reply by ghost** [reply]
I've looked through both of these already. The source code was very fun to look through (it's pretty cool, has a lot of stuff I haven't seen before, I learned a lot from it), but the docs were very sparse on how to actually modify and test the system. For example, if I wanted to change something in InstallHelloSystem.app, how would I do that? Also, how would I test it?

**Reply by probonopd** [reply, marked as answer]
Copy `Install helloSystem.app`, right-click, Show Contents, open the files in a text editor/IDE and hack away. :-)

**Reply by ghost** [reply]
Oh. It's that simple? I thought it would be more complicated than that haha. I don't think I'll need to actually test anything to the point of erasing my disk (which is what install hellosystem.app does iirc). I'll get started sometime soon (I want to play some COD before I switch OS). Thanks for helping!

**Reply by probonopd** [reply]
Yes, everything in helloSystem is designed to be simple. If you find something complicated, it's probably a bug and should be fixed ;-)
